src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -136,17 +127,70 @@ void CPlayer::drawDeathEffect(HexagonGame& mHexagonGame)
     }
 }
 
-void CPlayer::playerSwap(HexagonGame& mHexagonGame, bool mPlaySound)
-{
+void CPlayer::playerSwap(HexagonGame &mHexagonGame, bool mPlaySound) {
     angle += ssvu::pi;
     mHexagonGame.runLuaFunctionIfExists<void>("onCursorSwap"); // TODO: docs
 
-    if(mPlaySound)
-    {
+    if (mPlaySound) {
         mHexagonGame.getAssets().playSound("swap.ogg");
     }
 }
 
+[[nodiscard]] sf::Vector2f CPlayer::getPosition() const {
+    return pos;
+}
+
+[[nodiscard]] sf::Vector2f CPlayer::getStartPosition() const {
+    return startPos;
+}
+
+void CPlayer::kill(HexagonGame &mHexagonGame, bool mEffect) {
+    deadEffectTimer.restart();
+
+    if (!Config::getInvincible()) { dead = true; }
+    if (mEffect) { mHexagonGame.death(); }
+
+    ssvs::moveTowards(lastPos, ssvs::zeroVec2f, 5 * mHexagonGame.getSpeedMultDM());
+
+    pos = lastPos;
+}
+
+void CPlayer::push(HexagonGame& mHexagonGame, CWall& wall){
+    const auto curveData = wall.getCurve();
```

Comment:
```suggestion
    const auto& curveData = wall.getCurve();
```

No need to copy here.